packages/components/src/components/ViewYAML/ViewYAML.js

```diff
@@ -15,20 +15,23 @@ import React from 'react';
 import PropTypes from 'prop-types';
 import jsYaml from 'js-yaml';
 import classNames from 'classnames';
+import { Prism as SyntaxHighlight } from 'react-syntax-highlighter';
 
 const ViewYAML = props => {
-  const { className, dark, resource } = props;
+  const { className, dark, resource, enableSyntaxHighlighting = false } = props;
 
   return (
-    <div
+    <SyntaxHighlight
       className={classNames('bx--snippet--multi', className, {
-        'tkn--view-yaml--dark': dark
+        'tkn--view-yaml--dark': dark,
+        'tkn--view-yaml--enable-syntax-highlighting': enableSyntaxHighlighting
```

Comment:
no need for this line as the additional classes will only be added to the content when highlighting is enabled anyway

Reply:
Thank you for your review. This would be a good choice. ^^

Let me take a look tomorrow.

Reply:
@AlanGreene After I try, I found that we cannot replace the container element easily because the child structure is not the same. So I refactoring it to 2 render functions and select render function base on `enableSyntaxHighlighter` like this https://github.com/tektoncd/dashboard/pull/2071/files#diff-7d3c10c7cf56e92d73518e8412053ccc766adc892693aedc504249e8c78c50d5R23-R44. 

What do you think?